### PR TITLE
fix: use `matchSourceUrl` to match monorepo packages

### LINKED ISSUE #18899: matchSourceUrlPrefixes of monorepo:capacitor matches packages outside of this repo
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### If you're self-hosting Renovate, select which platform you are using.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I use the `monorepo:capacitor` monorepo [preset](https://docs.renovatebot.com/presets-monorepo/#monorepocapacitor). I have disabled this repo like this:

```json
    {
      "matchFiles": [
        "packages/app/package.json"
      ],
      "extends": [
        "monorepo:capacitor"
      ],
      "enabled": false
    },
```
This works fine, but [@capacitor/assets](https://www.npmjs.com/package/@capacitor/assets)  which has the repo https://github.com/ionic-team/capacitor-assets is not being updated, because its source URL has the same prefix as the the packages which have the repo https://github.com/ionic-team/capacitor. 

Why does the  `monorepo:capacitor` preset use `matchSourceUrlPrefixes` intead of `matchSourceUrl`?

### Relevant debug logs

<details><summary>Logs</summary>

```json
          {
            "depType": "devDependencies",
            "depName": "@capacitor/assets",
            "currentValue": "2.0.0",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "2.0.0",
            "depIndex": 47,
            "updates": [],
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/ionic-team/capacitor-assets",
            "homepage": "https://ionicframework.com",
            "currentVersion": "2.0.0",
            "fixedVersion": "2.0.0"
          },
```
But 2.0.1 is released.

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Context
- Closes: #18899 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required
- [x] I think the docs will be updated automatically 🤔

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
PR title seems misleading 🙃

**Comment by RahulGautamSingh** [comment]
Waiting on [this](https://github.com/renovatebot/renovate/issues/18899#issuecomment-1319413355)

File: lib/config/migration.spec.ts [hunk 1]
```diff
@@ -549,7 +549,7 @@ describe('config/migration', () => {
             packagePatterns: ['^bar'],
             excludePackageNames: ['baz'],
             excludePackagePatterns: ['^baz'],
-            sourceUrlPrefixes: ['https://github.com/vuejs/vue'],
+            sourceUrlPrefixes: ['https://github.com/lodash'],
```
Comment (RahulGautamSingh) [inline lib/config/migration.spec.ts:552]: Maybe unnecessary 🙃. Will let you guys decide

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -1908,42 +1908,42 @@ The above will match only the root `package.json`, whereas the following would m
...
   "packageRules": [
     {
-      "matchSourceUrlPrefixes": ["https://github.com/renovatebot/"],
-      "groupName": "All renovate packages"
+      "matchSourceUrls": ["https://github.com/facebook/react"],
+      "groupName": "React"
     }
   ]
 }
 ```
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1938]: I think delete this first example - no need for two

File: lib/config/migration.spec.ts (hunk 1 above)
Comment (rarkins) [reply lib/config/migration.spec.ts:552]: It's fine

File: lib/config/presets/internal/monorepo.ts [hunk 3]
```diff
@@ -240,6 +235,14 @@ const repoGroups = {
   'zxing-net': 'https://github.com/micjahn/ZXing.Net',
 };
 
+const orgGroups = {
+  hapijs: 'https://github.com/hapijs',
```
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:239]:
```suggestion
  hapijs: 'https://github.com/hapijs/',
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/presets/internal/monorepo.ts [hunk 4]
```diff
@@ -240,6 +235,14 @@ const repoGroups = {
   'zxing-net': 'https://github.com/micjahn/ZXing.Net',
 };
 
+const orgGroups = {
```
Comment (viceice) [inline lib/config/presets/internal/monorepo.ts:238]: please add the eslint comment for key order

**Review by viceice — COMMENTED** [review]
looks good so far. but we need to validate if this still groups right. eg sometimes the sourceUrl of a package points to a file or folder inside the repo. do we massage those urls correctly before we match against package rules?

**Comment by rarkins** [comment]
> looks good so far. but we need to validate if this still groups right. eg sometimes the sourceUrl of a package points to a file or folder inside the repo. do we massage those urls correctly before we match against package rules?

I guess we should double check the massaging we do in `datasource` metadata logic. These are all github repos, so we should be able to strip their `sourceUrls` to match.

**Comment by rarkins** [comment]
> > looks good so far. but we need to validate if this still groups right. eg sometimes the sourceUrl of a package points to a file or folder inside the repo. do we massage those urls correctly before we match against package rules?
> 
> I guess we should double check the massaging we do in `datasource` metadata logic. These are all github repos, so we should be able to strip their `sourceUrls` to match.

Looks like we do: https://github.com/renovatebot/renovate/blob/c45c8759202283ceb79bd5e853c75970728b8417/lib/modules/datasource/metadata.ts#L30

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]
🤞

**Pull request merged**